ExomeDepth/run_ExomeDepth.py

```diff
@@ -198,132 +204,174 @@ def multiprocess_call(multiprocess_list):
             )
 
     if args.vcf_filename_suffix:
-        action = "{action} --vcf_filename_suffix {vcf_filename_suffix}".format(action=action, vcf_filename_suffix=args.vcf_filename_suffix)
+        action = "{action} --vcf_filename_suffix {vcf_filename_suffix}".format(
+            action=action, vcf_filename_suffix=args.vcf_filename_suffix
+        )
 
     os.system(action)
 
+
+def add_sample_to_db(sample_id, flowcell_id, refset):
+    Session = connect_database()
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).all():
+            entry = create_sample(sample_id, flowcell_id, refset)
+            store_sample(Session, entry)
+            return True
+        else:
+            return False
+
+
+def query_refset(sample_id, flowcell_id):
+    Session = connect_database()
+    with Session() as session:
+        return session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+
+
 def call_cnv(args):
 
     """Call CNV from BAMs"""
     bam = format(os.path.abspath(args.inputbam))
     output_folder = format(os.path.abspath(args.output))
     analysis = settings.analysis
-    prob = settings.probability
+
     if not os.path.isdir(output_folder):
-         os.system("mkdir -p " + str(output_folder)) 
+        os.system("mkdir -p " + str(output_folder))
 
     """Determine gender"""
     if args.refset_gender:  # Used gender if used as input parameter.
         gender = args.refset_gender
     else:  # Otherwise determine based on chrY count
         gender = get_gender(bam)
 
-    multiprocess_list=[]
+    flowcell_id = get_flowcelid_bam(bam)
+
+    if args.refset:  # Do not query database to query refset
+        refset = args.refset
+    else:
+        """ Add sample to database if not present, or query refset from db if present """
+        refset = settings.refset  # Default refset in settings.py
+        if not add_sample_to_db(args.sample, flowcell_id, refset):
+            """ Query database if sample + flowcell present in database """
+            refset = query_refset(args.sample, flowcell_id)  # Get refset from database
+
+    multiprocess_list = []
     for model in analysis:
-        multiprocess_list += [[model, output_folder, gender, analysis[model]["target_bed"], analysis[model]["exon_bed"], bam, output_folder, analysis[model]["calling_model"]]] 
-    
+        multiprocess_list += [[
+            model, output_folder, gender, analysis[model]["target_bed"],
+            analysis[model]["exon_bed"], bam, output_folder, refset
+            ]
+        ]
```

Comment:
Complex nested list? Maybe used a list of dictionaries?

Reply:
I don't see the complexity here. 
What do you suggest as a list of dictionaries?

Reply:
It is complex because, it is long and order matters. It is used multiple times using different combinations of items in the list. I think that a dictionary would suite the usage if this list better.

Code would look like this:
```python
multiprocess_list.append({
    'model': model,
    'output_folder': output_folder,
   ...
})
```